components/client-pangolin-s2s/src/feemarket.rs

```diff
@@ -106,4 +111,67 @@ impl RelayStrategy for PangolinRelayStrategy {
         }
         false
     }
+
+    async fn callback<
+        P: MessageLane,
+        SourceClient: MessageLaneSourceClient<P>,
+        TargetClient: MessageLaneTargetClient<P>,
+    >(
+        &self,
+        reference: &RelayReference<P, SourceClient, TargetClient>,
+        decide: bool,
+    ) {
+        let mut total_cost = P::SourceChainBalance::zero();
+
+        let confirmation_transaction_cost = reference
+            .lane_source_client
+            .estimate_confirmation_transaction()
+            .await;
+
+        let delivery_transaction_cost = match reference
```

Comment:
3. Calculate the current transaction delivery cost from target chain, and convert the target chain price into the source chain price based on the actual exchange ratio